### ISSUE TITLE
Resolve dependencies off package.json directly (better shadow-cljs support)

### ISSUE THREAD

**Issue by binarykitchen** [cursive-ide/cursive#2333]
Has anyone ever managed to resolve npm dependencies installed by shadow-cljs in cursive? 

Cursive always displays these as 

_“x cannot be resolved”._ 

This because the npm dependencies are in package.json but not project.clj - Cursive seems to only scan those in project.clj. Thoughts?

PS: `shadow-cljs pom` and importing this via Maven didn't work

**Comment by thheller** [comment]
Note that `package.json` is completely irrelevant when it comes to resolving stuff. `shadow-cljs` never looks at it for the regular project. Only the contents of the `node_modules` folder are considered and not how they got there.

**Comment by conan** [comment]
npm deps don't seem to resolve, and it would be really useful if they did - I get a lot of "cannot be resolved" false positives for things required using shadow's syntax:
```clojure
["@material-ui/icons/Stars" :default Stars]
["@material-ui/icons" :as mui-icons]
["@material-ui/pickers" :refer [KeyboardDatePicker]]
```
In this example, cursive can resolve neither `Stars`, `mui-icons/Stars` nor `KeyboardDatePicker`

**Comment by Wegi** [comment]
Yes, especially when working with react-bundling apps like reagent / re-frame / om.next / fulcro this is an Issue often encountered. Our team would really appreciate a feature, that takes this into account.

**Comment by devurandom** [comment]
IntelliJ has a NodeJS plugin. Is it somehow possible for Cursive to hook into that?

**Comment by cursive-ide** [comment]
It's possible to hook into it's package.json support, yes, but that doesn't supply the exports from those packages. That is provided by the JS support, and that is very hard to integrate with. I've made several attempts, but it relies on the source looking very much like JS, unfortunately.

**Comment by ulisesmac** [comment]
@cursive-ide 
Maybe not actually resolved if it's so hard, but what about an option to treat them as recently defined symbols?

e.g.
```clojure
["react" :as react]
```
Would be the same as:
```clojure
(def react nil) ;; nil or whatever
```

and similar for 
```clojure
["@mui/material" :refer [CardActionArea]]
```

So that we don't get those annoying warnings.

This could be an option that can be toggled, I personally work a lot using ClojureScript, and it'd be really good to not mark them as warnings.
